File: WEBapiSPA/Model/Message.cs
namespace WEBapiSPA.Model
{
    public class Message
    {
        public Guid Id { get; set; }
        public Guid Device { get; set; }
        public string UserName { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string VersionPA { get; set; }

        public override string ToString()
        {
            return $"Id: {Id}; Device: {Device}; UserName: {UserName};\n" +
                $" StartTime: {StartTime}; EndTime: {EndTime}; VersionPA: {VersionPA};";
        }
    }
}

File: SPAAngular/src/app/message.ts
import { Data } from "@angular/router";

//export class Message {
//  public Id?: string;
//  public UserName?: string;
//  public StartTime?: Date;
//  public EndTime?: Date;
//  public VersionPA?: string;

//  constructor(
//    id?: string,
//    userName?: string,
//    startTime?: Date,
//    endTime?: Date,
//    versionPA?: string)
//  {
//    this.Id = id;
//    this.UserName = userName;
//    this.StartTime = startTime;
//    this.EndTime = endTime;
//    this.VersionPA = versionPA;
//  }
//}

export interface Message {
  id: string;
  userName: string;
  startTime: Date;
  endTime: Date;
  versionPA: string;
}
